sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -147,7 +147,9 @@ def _urljoin(base_url, stub_url):
     :rtype: str
     """
     parsed = urlparse(base_url)
-    parsed = parsed._replace(path=parsed.path + "/" + stub_url)
+    parsed = (parsed._replace(
+        path=parsed.path + "/" + stub_url) if parsed.path[-1:] != "/" else parsed._replace(
+        path=parsed.path + stub_url))
```

Comment:
You can make this simpler with:
```
parsed = parsed._replace(path=parsed.path.rstrip("/") + "/" + stub_url)
```
Which will remove the trailing "/" for `parsed.path` only if it exists.